docs/slurk_bots_events.rst

```diff
@@ -0,0 +1,228 @@
+                response = requests.post(
+                    f"{self.uri}/users/{self.user}/rooms/{room_id}",
+                    headers={"Authorization": f"Bearer {self.token}"}
+                )
+
+If a task room has been opened, it compares the task id of the task room with its own task id. It joins the room only if the two task ids match or it is not assigned any task id. We interpret the latter as the echo bot being relevant to all tasks.
+
+Movement
+~~~~~~~~
+Monitoring own movement
+-----------------------
+
+Bots can monitor their own movement between rooms. This includes them joining or leaving a room. For this purpose they have to listen to the ``joined_room`` and ``left_room`` events, respectively. Both events are only sent to the user (e.g. bot) that caused them.
+
+.. code-block:: python
+
+    @self.sio.event
+    def joined_room(data):
+        do_something(data)
+
+``data`` in ``joined_room`` has this structure:
+
+- ``user(int)``: the ``id`` of the user who caused this event
+- ``room(int)``: the ``id`` of the room that was entered by this user
+
+Task bots are generally sent to rooms to instruct users and provide resources necessary to the task fulfillment. The ``joined_room`` event handler could be used to introduce the bot to the users and set an initial task description.
+
+.. code-block:: python
+
+    @self.sio.event
+    def left_room(data):
+        do_something(data)
+
+``data`` in ``left_room`` has this structure:
+
+- ``user(int)``: the ``id`` of the user who caused this event
+- ``room(int)``: the ``id`` of the room that was left by this user
+
+Monitoring overall movement
+---------------------------
+Bots are also notified once a user joins or leaves one of the rooms the bot is placed in. The term `user` here includes the bot itself as well as other bots and human users.
+
+
+.. code-block:: python
+
+    @self.sio.event
+    def status(data):
+        do_something(data)
+
+``data`` in ``status`` has this structure:
+
+- ``type(str)``: the status type, either `join` or `leave`
+- ``user(dict)``: dictionary of ``id(int)`` and ``name(str)`` of the user who caused this event
+- ``room(int)``: the ``id`` of the room that was entered or left, respectively
+- ``timestamp(str)``: as ISO 8601: ``YYYY-MM-DD hh:mm:ss.ssssss`` in UTC Time
+
+Chat
+~~~~
+All of the events mentioned below can be either ``private`` or not. If an event is ``private`` it is only sent to a designated receiver. If this receiver is the bot, it receives the event. Otherwise it does not receive it. If an event is not ``private`` it can be seen by all users in the specified room.
+Only bots are intended to send private content but for debugging purposes you can use the chat interface and the following syntax to send private messages ``@<user_id> <text>`` or private images ``@<user_id> image: <url>``. Make sure that whoever is supposed to send private content is assigned the ``send_privately`` permission.

```

Comment:
```suggestion
All the events mentioned below can be ``private`` or not. If an event is ``private`` it is only sent to a designated receiver. If this receiver is the bot, it receives the event. Otherwise it does not receive it. If an event is not ``private`` it can be seen by all users in the specified room.
Only bots should send private content, but for debugging purposes, you can use the chat interface and the following syntax to send private messages ``@<user_id> <text>`` or private images ``@<user_id> image: <url>``. Make sure that whoever is supposed to send private content is assigned the ``send_privately`` permission.
```